### DISCUSSION TITLE
Block Rendering / Attribute Value Sync issues.

### DISCUSSION THREAD

**Discussion by Targoviste** [DomCR/ACadSharp#465] [Q&A] [answered]
First of all I'd just like to thank you and everyone else who has helped make this amazing tool!

I'm having 2 issues I'm having a hard time tracking down the possible cause for.

I am using the v1.0.0 NuGet package.

### **Problem 1:**  
I'm copying blocks from a reference DWG that have attributes. I am able to redefine the AttributeDefinitions and can verify they are are updated and exist in the saved out DWG using this tool. 

However opening the DWG in AutoCAD I'm unable to see the values in the model space. The individual Block Inserts do not show Attributes or Values in the AutoCAD Properties pane either.  

Inside AutoCAD using the Block Editor I am able to see the attribute definitions. 

It is only after running the ATTSYNC command on the block that the AttributeDefintions are rendered.

### **Problem 2:**
These blocks have Hatches inside of them that don't seem to be carried over properly upon cloning and insertion, is that a current limitation of the state of ACadSharp or something wrong on my implementation end?

            OpenFileDialog openFileDialog = new OpenFileDialog
            {
                Filter = "DWG files (*.dwg)|*.dwg",
                Title = "Select Reference DWG File"
            };

            if (openFileDialog.ShowDialog() == true)
            {
                string referenceDwgPath = openFileDialog.FileName;

                // Load the reference DWG
                using (DwgReader referenceDwgReader = new DwgReader(referenceDwgPath))
                {
                    CadDocument referenceDoc = referenceDwgReader.Read();

                    // Create a new DWG file
                    CadDocument newDwgDoc = new CadDocument();

                    // Create a new BlockRecord instance
                    BlockRecord newBlockRecord = (BlockRecord)referenceDoc.BlockRecords["DUCT_CENTER"].Clone();

                    // Add the cloned block record to the new document
                    newDwgDoc.BlockRecords.Add(newBlockRecord);

                    // Loop through each room type and room data
                    foreach (var roomType in RoomsByType.Keys)
                    {
                        foreach (var room in RoomsByType[roomType])
                        {
                            // Find the block that matches
                            if (newDwgDoc.BlockRecords.Contains("DUCT_CENTER"))
                            {
                                BlockRecord blockRecord = newDwgDoc.BlockRecords["DUCT_CENTER"];

                                foreach(AttributeDefinition attributeDefinition in blockRecord.AttributeDefinitions)
                                {
                                    if(attributeDefinition.Tag == "NOTE_1")
                                    {
                                        attributeDefinition.Value = "UPDATED_VALUE";
                                    }
                                }

                                Insert insert = new Insert(blockRecord)
                                {
                                    InsertPoint = new CSMath.XYZ(room.PositionX, room.PositionY, 0),
                                    XScale = 1 / 25.4, //Placeholder until scale factor check added
                                    YScale = 1 / 25.4,
                                    ZScale = 1 / 25.4,
                                };

                                // Add the cloned block reference to the model space
                                newDwgDoc.Entities.Add(insert);
                            }
                        }
                    }

                    // Save the new DWG file
                    SaveFileDialog saveFileDialog = new SaveFileDialog
                    {
                        Filter = "DWG files (*.dwg)|*.dwg",
                        Title = "Save Exported DWG File"
                    };

                    if (saveFileDialog.ShowDialog() == true)
                    {
                        using (DwgWriter writer = new DwgWriter(saveFileDialog.FileName, newDwgDoc))
                        {
                            writer.Write();
                        }

                        MessageBox.Show("Rooms exported to DWG successfully.", "Export Complete", MessageBoxButton.OK, MessageBoxImage.Information);
                    }
                }

**Comment by DomCR** [comment, marked as answer]
Hi @Targoviste,

For the problem 1, it has been solved:

- https://github.com/DomCR/ACadSharp/pull/449

I'll create the new release for 1.0.1 to fix the package.

For the problem 2 there is already an investigation going on to see what's happening:

- https://github.com/DomCR/ACadSharp/issues/458

Hope this helps! thanks for your support!
